### ISSUE TITLE
t.authorize() only resolves on Allow

### ISSUE THREAD

**Issue by smdern** [trello/api-docs#114]
t.authorize() doesn't complete the promise unless the user clicks 'Allow' which leaves a dangling promise.

It would be nice if the Promise would be rejected if the user clicks 'Deny' or closes the popup.

**Comment by bentleycook** [comment]
👋 Good catch. We're currently working on a bit more functionality that will give you the ability to just request a token from a user and return it to you in a Promise or a rejected Promise if they denied. Would that solve your use-case?

Our hope is that this makes it even easier and you don't have to do anything special aside from make a single request to the Trello Power-Up client library and expect back a token or a rejection.

**Comment by smdern** [comment]
> We're currently working on a bit more functionality that will give you the ability to just request a token from a user and return it to you in a Promise or a rejected Promise if they denied.

Yeah that would work. It would be nice to handle the 'Deny' case so that we could display some sort of 'This wont work unless you authorize' message.

**Comment by nickarora** [comment]
Just checking in on this.  This is still a concern for us -- any progress?

**Comment by bentleycook** [comment]
@nickarora Something we're still aware of and planning to work on.

Just to confirm, you're currently using `t.authorize()` to get a user's Trello token?
